packages/toast/src/toast.tsx

```diff
@@ -37,7 +37,7 @@ export const Toast: React.FC<ToastProps> = (props) => {
       width="auto"
       aria-labelledby={alertTitleId}
     >
-      <AlertIcon />
+      {icon || <AlertIcon />}
```

Comment:
By replacing the <AlertIcon /> with custom icon we remove the possibility to style it via the theme, because the theme styles are applied within the AlertIcon component.

https://github.com/chakra-ui/chakra-ui/blob/a26f7f8c02561f2a20cc6fac416fa5e758175eb1/packages/alert/src/alert.tsx#L122-L138

When we allow the AlertIcon to receive `children` we have the best out of both worlds.

```tsx
// packages/alert/src/alert.tsx
return (
    <chakra.span
      display="inherit"
      {...props}
      className={cx("chakra-alert__icon", props.className)}
      __css={css}
    >
      {props.children || <BaseIcon h="100%" w="100%" />}
    </chakra.span>
  )
```

and we can update the implementation here to

```diff
-{icon || <AlertIcon />}
+<AlertIcon>{icon}</AlertIcon>
```

What do you think?

Reply:
Yes, I think this kinda way the using custom icon is much more maintainable with the theme and stuff. Great suggestion 👍 .
Thanks 🤟🏻

Reply:
Hey, I have changed the code according to what you said.

Reply:
Awesome 🚀 
I think we still need the change in `alert.tsx`:

```diff
// packages/alert/src/alert.tsx
return (
    <chakra.span
      display="inherit"
      {...props}
      className={cx("chakra-alert__icon", props.className)}
      __css={css}
    >
-      <BaseIcon h="100%" w="100%" />
+      {props.children || <BaseIcon h="100%" w="100%" />}
    </chakra.span>
  )
```

Reply:
I'm sorry, Just forgot that  😅

Reply:
I have made the change, take a look 👀